vuepress/docs/next/tutorials/create/ms/add-access-controls.md

```diff
@@ -1,109 +1,161 @@
 
 ## Notes
 ### Realm Roles versus Client Authorities
-This tutorial made use of authorities which in Keycloak are Roles mapped to a User for a specific Client. You could also make use of higher-level Realm Roles assigned directly to users, e.g. `ROLE_ADMIN`. That will work but can result in collisions between applications if they happen to use the same roles.
+This tutorial utilizes authorities, which (in Keycloak) are roles mapped to a user for a specific client. It is possible to assign higher-level Realm Roles directly to users, e.g. `ROLE_ADMIN`, but this can result in collisions between applications using the same roles.
+
+To implement Realm-assigned roles, the code above must be modified:
+- In the backend, use the annotation `@Secured('ROLE_ADMIN)` or `@PreAuthorize(hasRole('ROLE_ADMIN'))`
+- In the frontend, use `keycloak.hasRealmRole` instead of `keycloak.hasResourceRole` 
+
+See the [Spring Security page](https://www.baeldung.com/spring-security-check-user-role) for more examples.
+
+### Local vs. Kubernetes Testing
+This tutorial leverages the `internal` client, which is configured in the microservice via the `application.yml`. Client roles are manually created and assigned in Keycloak. 
+
+In Kubernetes, Entando will automatically create client roles per the bundle plugin definition (see the [plugin definition](../../../docs/curate/ecr-bundle-details.md) for more information). These roles are created for the client specific to the microservice, e.g. `<docker username>-conference-server`. The client name is injected as an environment variable into the plugin container, so the annotations noted above will work in both local and Kubernetes environments.
 
-If you choose to use Realm-assigned roles then the code above would need to change. In the backend, use the following annotations: `@Secured('ROLE_ADMIN)` or `@PreAuthorize(hasRole('ROLE_ADMIN'))`. In the frontend, use `keycloak.hasRealmRole` instead of `keycloak.hasResourceRole`. See the [Spring Security page](https://www.baeldung.com/spring-security-check-user-role) for more examples.
+#### Modify Security Checks for Kubernetes
 
-### Local versus Kubernetes Testing
-This tutorial also makes use of the `internal` client configured in the microservice via the application.yml with roles manually created and assigned in Keycloak. In Kubernetes, Entando will automatically create client roles per the bundle plugin definition (see the plugin definition [here](../../../docs/curate/ecr-bundle-details.md) for more information). Those roles will be created for the client specific to the microservice itself, e.g. `<docker username>-conference-server`. This client name will be injected as an environment variable into the plugin container itself so the annotations noted above will work both in local and Kubernetes environments.
+The MFE authorization checks in this tutorial explicitly note the client ID,  e.g. `internal`. The following options modify the checks to work in Kubernetes:
```

Comment:
The MFE authorization checks in this tutorial explicitly note the client ID, e.g. `internal`.  >> The MFE authorization checks explicitly note the client ID in this tutorial, e.g... 

the sentence is a bit awkward and confusing >> Move 'in this tutorial' or remove altogether.

Reply:
tacking it on the end is awkward; i think it's important to confine the scope to the current tutorial. will move to front.